pkg/prometheus/promcfg.go

```diff
@@ -907,7 +914,11 @@ func (cg *ConfigGenerator) generateProbeConfig(
 	}
 
 	assetKey := fmt.Sprintf("probe/%s/%s", m.Namespace, m.Name)
-	cfg = addOAuth2ToYaml(cfg, version, m.Spec.OAuth2, oauth2Secrets, assetKey)
+	cfg = addOAuth2ToYaml(cfg, version, m.Spec.OAuth2, store.OAuth2Assets, assetKey)
+
+	if version.GTE(semver.MustParse("2.26.0")) {
+		cfg = addAuthorizationToYaml(cfg, fmt.Sprintf("probe/auth/%s/%s", m.Namespace, m.Name), store, m.Spec.Authorization)
```

Comment:
`auth` can be a bit ambiguous and is often used for authentication. To make the intent more explicit, we could use either `authz` or `authorization` in the key.

Reply:
Yeah you are correct, however I figured this is an internal implementation detail that a user won't ever see. Also the naming is off anyway, no component does any actual authorization :)

We could go with `authorization` to be very explicit, but that gets kinda long. I'm not feeling very strongly about this either way.

Reply:
Yeah it's definitely just a suggestion, I am also fine with leaving as it is. As you pointed out, it's an implementation detail which should not affect end-users.